src/military/military.cpp

```diff
@@ -3961,46 +3968,24 @@ void army_arrives_in_province(sys::state& state, dcon::army_id a, dcon::province
 		}
 
 		// start battle
-		dcon::land_battle_id gather_to_battle;
-		dcon::war_id battle_in_war;
-
+		auto gather_to_battle = dcon::land_battle_id{};
+		auto battle_in_war = dcon::war_id{};
```

Comment:
This is fine, but it is redundant because their default constructor already does this